lib/index.js

```diff
@@ -85,6 +85,20 @@ function Socket (domain, type) {
     this.binding = nn.Socket(this.af_domain, this.protocol);
     this.queue = [];
 
+    /**
+     * sockopts API
+     */
+     this.linger     = linger
+     this.sndbuf     = sndbuf
+     this.rcvbuf     = rcvbuf
+     this.sndtimeo   = sndtimeo
+     this.rcvtimeo   = rcvtimeo
+     this.reconn     = reconn
+     this.maxreconn  = maxreconn
+     this.sndprio    = sndprio
+     this.rcvprio    = rcvprio
+     this.tcpnodelay = tcpnodelay
```

Comment:
I'll have to be more stringent about style throughout the code base, but please add semicolons.